Publish to NPM and Github Pages

rschamp opened the pull request:
Make the playground rely only on files within its directory so that it can be published standalone to gh-pages.

Publish to NPM by creating an .npmrc file with an authentication token pointing to the encrypted `NPM_TOKEN` environment variable stored in Travis settings.

Remove the checked-in built assets, as they're now built before it's published to NPM. When installing from git, they will be built on npm install.

To publish to gh-pages, there is an environment variable `GH_TOKEN`, which stores a Personal Access Token I generated. If this token stops working, it can be updated in the Settings panel for the repo in Travis.

Commit:
Provide playground dependencies with webpack

This confines all the files the playground needs to the playground directory.

Commit:
Publish playground to gh-pages

Commit:
Publish to NPM

Authenticate with NPM via .npmrc and publish the package version + "-[git sha]".

Remove vm and vm.min.js, they'll be built before the package is published to NPM and available on installation. When installing from git, the build step will run on npm install.

Commit:
Add explanatory comments

tmickel commented on webpack.config.js:
```diff
@@ -23,23 +22,80 @@ var base = {
+    // Playground
+    defaultsDeep({}, base, {
+        entry: {
+            'playground/vm': './src/index.js',
```
nice!

tmickel commented on package.json:
```diff
@@ -11,24 +11,27 @@
-    "scratch-blocks": "git+https://git@github.com/LLK/scratch-blocks.git#develop",
-    "scratch-render": "git+https://git@github.com/LLK/scratch-render.git#develop",
+    "scratch-blocks": "0.1.0-80849",
+    "scratch-render": "0.1.0-9f26a",
```
btw...fixing these to a published version number is interesting. we can still use `npm link` for development, but people will get an outdated version by default if they just clone?

rschamp replied:
They will... although I think if we changed this to `^0.1.0` or just `*`, then we could have the most recent version on npm install.

tmickel replied:
Yeah - I might prefer that for new downloaders but don't strongly care, since we should be doing `npm link` anyway.

rschamp replied:
Done

rschamp commented:
Will need to resolve #219 before merging:

> http://llk.github.io/scratch-vm/
> The green-flag and repeat-arrow icons are showing for me as the "image missing" icon
> ![screen shot 2016-09-25 at 5 13 19 pm](https://cloud.githubusercontent.com/assets/2879020/18818944/6cfd406a-8343-11e6-89e5-be9b78abcc06.png)

rschamp commented:
The error on the playground `Uncaught TypeError: Cannot read property 'isConnected' of null` exists in the current develop branch playground.

Commit:
Update dev server to only serve the playground

Now that we don't rely on node_modules, we can update the content base to just `playground` and just look at it from http://localhost:8080/

rschamp commented:
@tmickel @thisandagain I updated one last thing — now that the playground doesn't rely on paths like `../node_modules/...`, we don't need to serve the entire project directory. So now the playground is served from `./playground` and is available at just `http://localhost:8080`.

tmickel commented:
Sounds good to me!

Commit:
Use most recent version of scratch-* dependencies

thisandagain commented:
This looks great. 👍

Commit:
Update version format

-prerelease.[timestamp] is semver friendly. Move SHA info to package.json.

Commit:
Use new version format for scratch-* dependencies
